core/domain/ui/blocks/event-attendees.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Template for EventAttendees block render
+ * Template variables passed in:
+ *
+ * @type EE_Attendee[] $attendees
+ * @type array         $attributes  Original attributes saved for the block.
+ * @throws EE_Error
+ */
+?>
+<div id="ee-block-event-attendees" class="ee-core-blocks event-espresso-blocks event-attendees">
+    <ul>
+        <?php
+        foreach ($attendees as $attendee) {
+            $gravatar = $attributes['showGravatar']
+                ? get_avatar_url(
+                    $attendee->email(),
+                    array(
+                        'default' => 'mm',
+                        'width'   => $attributes['avatarSize'],
+                        'height'  => $attributes['avatarSize']
+                    )
+                )
+                : '';
+            $gravatar_class = $attributes['avatarClass']
+                ? $attributes['avatarClass'] . ' contact-avatar-img avatar'
+                : 'contact-avatar-img avatar';
+            $gravatar = $gravatar !== ''
+                ? '<div class="contact-image-wrap-div">'
+                    . '<img class="' . $gravatar_class . '"'
+                    . ' width="' . $attributes['avatarSize'] . '"'
+                    . ' height="' . $attributes['avatarSize'] . '"'
+                    . ' src="' . $gravatar . '" alt="contact avatar">'
+                    . '</div>'
+                : '';
+            echo "<li>{$gravatar}<span>{$attendee->full_name()}</span>";
+        }
+        ?>
+    </ul>
+</div>
```

Comment:
In DDD the UI layer should be a separate layer all on its own, and can even be split up to accommodate different kinds of UIs.
Ideally the Domain layer only contains all of your application's raw business logic, which can then be accessed via various kinds of UIs and/or request types such as web browser, RSS feed, REST API, CLI, AJAX, Cron, etc...

So based on the above, and the fact that we already have a `/ui/browser/` folder, something like the following would be better:

```php
- /domain/
- /ui/
- /ui/cli/
- /ui/browser/
- /ui/browser/blocks
```

we have not done a good job of this at all and have waaaaay too much logic everywhere that has presentation layer type stuff embedded in both domain and application logic, but it helps to remember to try and separate those things as much as possible because then your domain logic is far more reusable (and functional) when it is free of presentation logic.